"""
@generated by mypy-protobuf.  Do not edit manually!
isort:skip_file
"""
import builtins
import google.cloud.websecurityscanner.v1.scan_config_error_pb2
import google.protobuf.descriptor
import google.protobuf.internal.enum_type_wrapper
import google.protobuf.message
import typing
import typing_extensions

DESCRIPTOR: google.protobuf.descriptor.FileDescriptor = ...

class ScanRunErrorTrace(google.protobuf.message.Message):
    """Output only.
    Defines an error trace message for a ScanRun.
    """
    DESCRIPTOR: google.protobuf.descriptor.Descriptor = ...
    class _Code:
        ValueType = typing.NewType('ValueType', builtins.int)
        V: typing_extensions.TypeAlias = ValueType
    class _CodeEnumTypeWrapper(google.protobuf.internal.enum_type_wrapper._EnumTypeWrapper[_Code.ValueType], builtins.type):
        DESCRIPTOR: google.protobuf.descriptor.EnumDescriptor = ...
        CODE_UNSPECIFIED: ScanRunErrorTrace.Code.ValueType = ...  # 0
        """Default value is never used."""

        INTERNAL_ERROR: ScanRunErrorTrace.Code.ValueType = ...  # 1
        """Indicates that the scan run failed due to an internal server error."""

        SCAN_CONFIG_ISSUE: ScanRunErrorTrace.Code.ValueType = ...  # 2
        """Indicates a scan configuration error, usually due to outdated ScanConfig
        settings, such as starting_urls or the DNS configuration.
        """

        AUTHENTICATION_CONFIG_ISSUE: ScanRunErrorTrace.Code.ValueType = ...  # 3
        """Indicates an authentication error, usually due to outdated ScanConfig
        authentication settings.
        """

        TIMED_OUT_WHILE_SCANNING: ScanRunErrorTrace.Code.ValueType = ...  # 4
        """Indicates a scan operation timeout, usually caused by a very large site."""

        TOO_MANY_REDIRECTS: ScanRunErrorTrace.Code.ValueType = ...  # 5
        """Indicates that a scan encountered excessive redirects, either to
        authentication or some other page outside of the scan scope.
        """

        TOO_MANY_HTTP_ERRORS: ScanRunErrorTrace.Code.ValueType = ...  # 6
        """Indicates that a scan encountered numerous errors from the web site
        pages. When available, most_common_http_error_code field indicates the
        most common HTTP error code encountered during the scan.
        """

    class Code(_Code, metaclass=_CodeEnumTypeWrapper):
        """Output only.
        Defines an error reason code.
        Next id: 7
        """
        pass

    CODE_UNSPECIFIED: ScanRunErrorTrace.Code.ValueType = ...  # 0
    """Default value is never used."""

    INTERNAL_ERROR: ScanRunErrorTrace.Code.ValueType = ...  # 1
    """Indicates that the scan run failed due to an internal server error."""

    SCAN_CONFIG_ISSUE: ScanRunErrorTrace.Code.ValueType = ...  # 2
    """Indicates a scan configuration error, usually due to outdated ScanConfig
    settings, such as starting_urls or the DNS configuration.
    """

    AUTHENTICATION_CONFIG_ISSUE: ScanRunErrorTrace.Code.ValueType = ...  # 3
    """Indicates an authentication error, usually due to outdated ScanConfig
    authentication settings.
    """

    TIMED_OUT_WHILE_SCANNING: ScanRunErrorTrace.Code.ValueType = ...  # 4
    """Indicates a scan operation timeout, usually caused by a very large site."""

    TOO_MANY_REDIRECTS: ScanRunErrorTrace.Code.ValueType = ...  # 5
    """Indicates that a scan encountered excessive redirects, either to
    authentication or some other page outside of the scan scope.
    """

    TOO_MANY_HTTP_ERRORS: ScanRunErrorTrace.Code.ValueType = ...  # 6
    """Indicates that a scan encountered numerous errors from the web site
    pages. When available, most_common_http_error_code field indicates the
    most common HTTP error code encountered during the scan.
    """


    CODE_FIELD_NUMBER: builtins.int
    SCAN_CONFIG_ERROR_FIELD_NUMBER: builtins.int
    MOST_COMMON_HTTP_ERROR_CODE_FIELD_NUMBER: builtins.int
    code: global___ScanRunErrorTrace.Code.ValueType = ...
    """Output only. Indicates the error reason code."""

    @property
    def scan_config_error(self) -> google.cloud.websecurityscanner.v1.scan_config_error_pb2.ScanConfigError:
        """Output only. If the scan encounters SCAN_CONFIG_ISSUE error, this field has the error
        message encountered during scan configuration validation that is performed
        before each scan run.
        """
        pass
    most_common_http_error_code: builtins.int = ...
    """Output only. If the scan encounters TOO_MANY_HTTP_ERRORS, this field indicates the most
    common HTTP error code, if such is available. For example, if this code is
    404, the scan has encountered too many NOT_FOUND responses.
    """

    def __init__(self,
        *,
        code : global___ScanRunErrorTrace.Code.ValueType = ...,
        scan_config_error : typing.Optional[google.cloud.websecurityscanner.v1.scan_config_error_pb2.ScanConfigError] = ...,
        most_common_http_error_code : builtins.int = ...,
        ) -> None: ...
    def HasField(self, field_name: typing_extensions.Literal["scan_config_error",b"scan_config_error"]) -> builtins.bool: ...
    def ClearField(self, field_name: typing_extensions.Literal["code",b"code","most_common_http_error_code",b"most_common_http_error_code","scan_config_error",b"scan_config_error"]) -> None: ...
global___ScanRunErrorTrace = ScanRunErrorTrace
